caliko/src/main/java/au/edu/federation/caliko/FabrikChain2D.java

```diff
@@ -290,7 +280,6 @@ public FabrikChain2D(FabrikChain2D source)
 		
 		// Native copy by value for primitive members
 		mChainLength            = source.mChainLength;
-		mNumBones               = source.mNumBones;
```

Comment:
You've decided you hate mNumBones and want to instead call mChain.size() --- okay, as you wish.

Surely calling mChain.size() ONCE and assigning it to something and then using that value is quicker that calling it every single time you want the value which is many times...

Stop "optimising" this code - if it goes slow we can optimise it. It doesn't go slow - so stop optimising it, please.